File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    wordlist = set()
    wordlist.add(words[0])
    
    for i in range(1, len(words)):
        current = words[i]
        prev_word = words[i - 1]
        
        if prev_word[-1] != current[0] or current in wordlist:
            return [i % n + 1, i // n + 1]
        
        wordlist.add(current)
    
    return [0, 0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <unordered_set>

using namespace std;

vector<int> solution(int n, vector<string> words) {
    unordered_set<string> wordlist;
    wordlist.insert(words[0]);
    
    for (int i = 1; i < words.size(); i++) {
        string current = words[i];
        string prev_word = words[i - 1];
        
        if (prev_word.back() != current[0] || wordlist.find(current) != wordlist.end()) {
            return {i % n + 1, i / n + 1};
        }
        
        wordlist.insert(current);
    }
    
    return {0, 0};
}